### PR TITLE
Fixing library dependency version

### PR DESCRIPTION
**Pull request by cochi2**
There is an issue where the released version is not installing in Maven. 
This is because an automated PR changed the `azure-communication-common` dependency version to a beta unreleased one, and this was not caught during the implementation.
This change aims to fix the dependency version.

Related: 
- https://github.com/Azure/azure-sdk-for-java/pull/22628
- https://github.com/Azure/azure-sdk-for-java/pull/23167
- https://github.com/Azure/azure-sdk-for-java/issues/23158

### CODE REVIEW

File: eng/jacoco-test-coverage/pom.xml [hunk 1]
```diff
@@ -64,7 +64,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-communication-common</artifactId>
-      <version>1.1.0-beta.1</version> <!-- {x-version-update;com.azure:azure-communication-common;current} -->
+      <version>1.0.3</version> <!-- {x-version-update;com.azure:azure-communication-common;dependency} -->
```
Comment (weshaggard) [inline eng/jacoco-test-coverage/pom.xml:67]: Why do you need to change the version here?

Comment (cochi2) [reply eng/jacoco-test-coverage/pom.xml:67]: I'd have thought that the versions should match, but if that's not the case I can revert to its original values.

Comment (weshaggard) [reply eng/jacoco-test-coverage/pom.xml:67]: I don't believe you need this change here. The other changes look correct and should be enough,

Comment (cochi2) [reply eng/jacoco-test-coverage/pom.xml]: Removed.

**Review by weshaggard — APPROVED** [review]

**Review by srnagar — APPROVED** [review]

**Pull request merged**